### PR TITLE
refactor: Top Banner and add google calendar credential banner

### LINKED ISSUE #12473: [CAL-2744] Email and Banners after google calendar integration has been expired/not working
[https://threads.com/thread/34547342300/34547342304?s=j8Gxi4iRV9rM2RwhVHtaJh](https://threads.com/thread/34547342300/34547342304?s=j8Gxi4iRV9rM2RwhVHtaJh)

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-2744](https://linear.app/calcom/issue/CAL-2744/email-and-banners-after-google-calendar-integration-has-been)</sub>

### PR DESCRIPTION
**Pull request by Udit-takkar**
Fixes: https://github.com/calcom/cal.com/issues/12473

TODO:
- [x] Fix Type error

Screenshot 2023-12-02 at 12 47 19 AM

Screenshot 2023-12-02 at 12 47 39 AM

### CODE REVIEW

File: packages/trpc/server/routers/loggedInViewer/getUserTopBanners.handler.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  let doesInvalidCredentialsExist = false;
+
+  for (const calendar of connectedCalendars) {
+    const isErrorPresent = !!calendar.error;
+    if (isErrorPresent) {
+      doesInvalidCredentialsExist = true;
+    }
+  }
+
+  return doesInvalidCredentialsExist;
```
Comment (ThyMinimalDev) [inline packages/trpc/server/routers/loggedInViewer/getUserTopBanners.handler.ts:42]:
```suggestion
  return connectedCalendars.some(calendar => !!calendar.error );
```

File: packages/trpc/server/routers/viewer/organizations/checkIfOrgNeedsUpgrade.handler.ts [hunk 2]
```diff
@@ -12,7 +11,7 @@ type GetUpgradeableOptions = {
 };
 
 export async function checkIfOrgNeedsUpgradeHandler({ ctx }: GetUpgradeableOptions) {
-  if (!IS_TEAM_BILLING_ENABLED) return [];
+  // if (!IS_TEAM_BILLING_ENABLED) return [];
```
Comment (ThyMinimalDev) [inline packages/trpc/server/routers/viewer/organizations/checkIfOrgNeedsUpgrade.handler.ts:14]: do we need to keep this ?

File: packages/trpc/server/routers/viewer/teams/getUpgradeable.handler.ts [hunk 3]
```diff
@@ -11,7 +10,7 @@ type GetUpgradeableOptions = {
 };
 
 export const getUpgradeableHandler = async ({ ctx }: GetUpgradeableOptions) => {
-  if (!IS_TEAM_BILLING_ENABLED) return [];
+  // if (!IS_TEAM_BILLING_ENABLED) return [];
```
Comment (ThyMinimalDev) [inline packages/trpc/server/routers/viewer/teams/getUpgradeable.handler.ts:13]: same do we need to keep this ?

File: packages/trpc/server/routers/viewer/organizations/checkIfOrgNeedsUpgrade.handler.ts (hunk 2 above)
Comment (Udit-takkar) [reply packages/trpc/server/routers/viewer/organizations/checkIfOrgNeedsUpgrade.handler.ts:14]: I'll undo this. I was just testing this banner

File: packages/trpc/server/routers/viewer/teams/getUpgradeable.handler.ts (hunk 3 above)
Comment (Udit-takkar) [reply packages/trpc/server/routers/viewer/teams/getUpgradeable.handler.ts:13]: I'll undo this. I was just testing this banner

**Review by ThyMinimalDev — DISMISSED** [review]
LGTM

**Comment by CarinaWolli** [comment]
Type check is failing, can you take a look @Udit-takkar

**Review by zomars — CHANGES_REQUESTED** [review]
Hey @Udit-takkar it seems we have a type error 

image

File: packages/ui/components/top-banner/TopBanner.tsx [hunk 4]
```diff
@@ -41,7 +42,7 @@ export function TopBanner(props: TopBannerProps) {
     <div
       data-testid="banner"
       className={classNames(
-        "flex min-h-[40px] w-full items-start justify-between gap-8 px-4 py-2 text-center lg:items-center",
+        `flex min-h-[${TOP_BANNER_HEIGHT}px] w-full items-start justify-between gap-8 px-4 py-2 text-center lg:items-center`,
```
Comment (zomars) [inline packages/ui/components/top-banner/TopBanner.tsx:45]: AFAIK this is not a recommended practice from the tailwindcss team. Here are the docs on it: https://tailwindcss.com/docs/content-configuration#dynamic-class-names

**Review by zomars — APPROVED** [review]

**Pull request merged**